swarm/chunk/chunk.go

```diff
@@ -0,0 +1,5 @@
+package chunk
+
+const (
+	DefaultSize int64 = 4096
```

Comment:
It is not an overkill, it is fine.

Reply:
Not to be mis-understood: Ideally `bmt` should not depend on `chunk` package. `bmt` should be stand-alone, maybe other people would want to use just it.

However the current way we set it up is confusing and not ideally for our purposes.
1. `hasher` is always `sha3.NewKeccak256` in our codebase. This does not have to be a param to `NewTreePool`.
2. We seem to be calculating the chunk size within the `NewTreePool` constructor - `segmentCount * segmentSize` - why not specify the chunk size as a param and infer `segmentCount` from it, considering that `segmentSize == hasher.Size()`.

I think this API can be improved, so that we drop the `chunk` dep that I have just included.